cli/kaos_cli/services/tests/test_state_service.py

```diff
@@ -90,139 +98,178 @@ def state_service_test_inputs():
     ]
 
 
+@parameterized_class(('context', 'param'), state_service_test_inputs())
 class TestStateService(TestCase):
 
-    # Arrange
-    TestCase.state_service = StateService()
-    TestCase.switch = False
+    def instantiate_service(self):
+        # Arrange
+        self.state_service = StateService()
+        self.switch = False
```

Comment:
I don't want to sounds pedantic, but when tests maintain state, they get complex and this pattern enable to maintain more state in the future, a more functional implementation of this method as a class-method that returns new values is more suitable for unit tests.

There is no need for the maintenance of state at the level of the `TestStateService` class instance, what you need is a new `StateService` instance and a `False` boolean every time you call this method from the unit test, let's keep the implementation at the level that it's needed.

Also, please document this method, what does the `switch` value signify, how should this method be used?

Reply:
Well, this switch only made sense earlier when the tests shared a common state (this was done in order to increase reusability although not the best practice). This has now been removed as the states are independent in each test case